sunscreen_tfhe/src/entities/glwe_secret_key.rs

```diff
@@ -72,6 +72,37 @@ where
     pub fn generate_uniform(params: &GlweDef) -> GlweSecretKey<S> {
         Self::generate(params, || uniform_torus::<S>().inner())
     }
+
+    /// Generate a binary GLWE secret key with a specific seed for deterministic generation.
+    pub fn generate_binary_with_seed(params: &GlweDef, seed: &Seed) -> GlweSecretKey<S> {
+        GlweSecretKey::generate_binary_with_rng(params, &mut seed.create_rng())
+    }
+
+    /// Generate a uniform GLWE secret key with a specific seed for deterministic generation.
+    pub fn generate_uniform_with_seed(params: &GlweDef, seed: &Seed) -> GlweSecretKey<S> {
+        params.assert_valid();
+
+        let len = GlweSecretKeyRef::<S>::size(params.dim);
+        let mut rng = seed.create_rng();
+
+        GlweSecretKey {
+            data: avec_from_iter!((0..len).map(|_| uniform_torus_with_seed::<S>(&mut rng).inner())),
+        }
+    }
+
+    /// Generate a binary GLWE secret key using a provided mutable RNG.
+    pub fn generate_binary_with_rng(
+        params: &GlweDef,
+        rng: &mut crate::rand::SeededRng,
```

Comment:
This function is not intrinsically secure because a user could use an insecure RNG

Reply:
I can make it private; the purpose is to cascade the use of a ChaCha RNG into generating two keys. Also the `SeededRng` is an alias for a CSPRNG.